app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java

```diff
@@ -89,11 +93,28 @@ public void forceUnsubscribe() {
         }
     }
 
-    public void goToNewsDetail(News news) {
-        Intent intent = new Intent(getActivity(), NewsDetailActivity.class);
-        intent.putExtra(Constants.NEWS, Parcels.wrap(news));
-
-        startActivity(intent);
+    public void goToNewsDetail(final News news) {
+        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(getActivity());
+
+        if (pref.getBoolean(Constants.CONFIG_EXTERNAL_BROWSER, false)) {
+            Intent intent = new Intent(getActivity(), NewsCommentsActivity.class);
+            intent.putExtra(Constants.NEWS, Parcels.wrap(news));
```

Comment:
Can't we extract these duplicate `intent.putExtra` and `startActivity(intent)` to outside the function? (line 101,102 and 115,116)